analyses/telomerase-activity-prediction/util/boxplot_by_molecular_subtype.R

```diff
@@ -0,0 +1,44 @@
+suppressPackageStartupMessages({
+  library(ggpubr)
+  library(ggplot2)
+  library(tidyverse)
+})
+
+theme_set(theme_classic() +
```

Comment:
yes, @komalsrathi I am going to create a figures README for theme and statistics guidance, but you can see what I had chosen in the TP53 module here: https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/2208e46df86046a6f88470cbd4a03ad3220d2597/analyses/tp53_nf1_score/05-tp53-altered-annotation.Rmd#L354-L392 - `theme_pubr() and palette = "jco"` for starters. For the statistical comparisons, we will want to use the non-parametric `kruskal.test` and do a correction using `bonferroni`.  I also added a snippet for printing the group Ns in the html file (we could also consider adding those to the figure x-axis in parentheses). Mentioning in case you want to work on this before I am able to get the README together.

Reply:
@jharenza A few issues I am facing with your suggested changes:

1. When I have > 10 categories, the jco palette throws a warning and does not show any boxes beyond 10. To avoid this, I am filtering: molecular_subtype == `NA`, or anything with `To be classified`, and filtering to sample count >=5

```
# before
> plyr::count(telomerase_scores, c("broad_histology", "molecular_subtype"))
                                 broad_histology                        molecular_subtype freq
1                                   Benign tumor                                     <NA>   33
2                                       Chordoma                       CHDM, conventional    2
3                                       Chordoma              CHDM, poorly differentiated    4
4                           Choroid plexus tumor                                     <NA>    5
5  Diffuse astrocytic and oligodendroglial tumor                              DMG, H3 K28   10
6  Diffuse astrocytic and oligodendroglial tumor              DMG, H3 K28, TP53 activated    3
7  Diffuse astrocytic and oligodendroglial tumor                   DMG, H3 K28, TP53 loss   21
8  Diffuse astrocytic and oligodendroglial tumor                              HGG, H3 G35    4
9  Diffuse astrocytic and oligodendroglial tumor                   HGG, H3 G35, TP53 loss    1
10 Diffuse astrocytic and oligodendroglial tumor                         HGG, H3 wildtype   32
11 Diffuse astrocytic and oligodendroglial tumor         HGG, H3 wildtype, TP53 activated    5
12 Diffuse astrocytic and oligodendroglial tumor              HGG, H3 wildtype, TP53 loss   18
13 Diffuse astrocytic and oligodendroglial tumor                      HGG, IDH, TP53 loss    1
14 Diffuse astrocytic and oligodendroglial tumor                    HGG, To be classified   41
15 Diffuse astrocytic and oligodendroglial tumor         HGG, To be classified, TP53 loss    1
16 Diffuse astrocytic and oligodendroglial tumor                                     <NA>    2
17                               Embryonal tumor                       CNS Embryonal, NOS   10
18                               Embryonal tumor                            CNS HGNET-MN1    1
19                               Embryonal tumor                             CNS NB-FOXR2    3
20                               Embryonal tumor                      ETMR, C19MC-altered    4
21                               Embryonal tumor                                ETMR, NOS    1
22                               Embryonal tumor                               MB, Group3   14
23                               Embryonal tumor                               MB, Group4   50
24                               Embryonal tumor                                  MB, SHH   30
25                               Embryonal tumor                     MB, To be classified   17
26                               Embryonal tumor                                  MB, WNT   10
27                               Embryonal tumor                                     <NA>   35
28                               Ependymal tumor                              EPN, H3 K28    1
29                               Ependymal tumor                                EPN, PF A    3
30                               Ependymal tumor                             EPN, ST RELA   29
31                               Ependymal tumor                             EPN, ST YAP1    4
32                               Ependymal tumor                    EPN, To be classified   56
33                               Germ cell tumor                                     <NA>   13
34                             Histiocytic tumor                                     <NA>    7
35                    Low-grade astrocytic tumor                          GNG, BRAF V600E   12
36                    Low-grade astrocytic tumor                GNG, BRAF V600E, CDKN2A/B    1
37                    Low-grade astrocytic tumor                                GNG, FGFR    1
38                    Low-grade astrocytic tumor                                  GNG, H3    1
39                    Low-grade astrocytic tumor                                 GNG, IDH    2
40                    Low-grade astrocytic tumor                       GNG, KIAA1549-BRAF    5
41                    Low-grade astrocytic tumor                           GNG, MYB/MYBL1    1
42                    Low-grade astrocytic tumor                        GNG, NF1-germline    1
43                    Low-grade astrocytic tumor                          GNG, other MAPK    4
44                    Low-grade astrocytic tumor                     GNG, other MAPK, IDH    1
45                    Low-grade astrocytic tumor                                 GNG, RTK    3
46                    Low-grade astrocytic tumor                    GNG, To be classified    1
47                    Low-grade astrocytic tumor                            GNG, wildtype   14
48                    Low-grade astrocytic tumor                          LGG, BRAF V600E   26
49                    Low-grade astrocytic tumor                LGG, BRAF V600E, CDKN2A/B    5
50                    Low-grade astrocytic tumor                                LGG, FGFR    8
51                    Low-grade astrocytic tumor                                 LGG, IDH    3
52                    Low-grade astrocytic tumor                       LGG, KIAA1549-BRAF  114
53                    Low-grade astrocytic tumor           LGG, KIAA1549-BRAF, other MAPK    1
54                    Low-grade astrocytic tumor                           LGG, MYB/MYBL1    2
55                    Low-grade astrocytic tumor                        LGG, NF1-germline    6
56                    Low-grade astrocytic tumor              LGG, NF1-germline, CDKN2A/B    1
57                    Low-grade astrocytic tumor                  LGG, NF1-germline, FGFR    2
58                    Low-grade astrocytic tumor                         LGG, NF1-somatic    2
59                    Low-grade astrocytic tumor                   LGG, NF1-somatic, FGFR    1
60                    Low-grade astrocytic tumor LGG, NF1-somatic, NF1-germline, CDKN2A/B    1
61                    Low-grade astrocytic tumor                          LGG, other MAPK   12
62                    Low-grade astrocytic tumor                                 LGG, RTK   12
63                    Low-grade astrocytic tumor                       LGG, RTK, CDKN2A/B    1
64                    Low-grade astrocytic tumor                    LGG, To be classified   10
65                    Low-grade astrocytic tumor                            LGG, wildtype   42
66                                      Lymphoma                                     <NA>    1
67                             Melanocytic tumor                                     <NA>    1
68                                    Meningioma                                     <NA>   29
69          Mesenchymal non-meningothelial tumor                                      EWS   11
70          Mesenchymal non-meningothelial tumor                                     <NA>   14
71                             Metastatic tumors                                     <NA>    7
72       Neuronal and mixed neuronal-glial tumor                                      CNC    2
73       Neuronal and mixed neuronal-glial tumor                                      EVN    1
74       Neuronal and mixed neuronal-glial tumor                                     <NA>   38
75                                 Non-CNS tumor                                     <NA>    1
76                                     Non-tumor                                     <NA>    3
77                        Other astrocytic tumor                                     <NA>    3
78                                   Other tumor                                     <NA>    1
79                          Pre-cancerous lesion                                     <NA>   14
80        Tumor of cranial and paraspinal nerves                                     <NA>   44
81                        Tumor of pineal region                                     <NA>    5
82                       Tumors of sellar region                             CRANIO, ADAM   24
83                       Tumors of sellar region                 CRANIO, To be classified   11

# After removing molecular subtype NA or To be classified and number of samples < 5
> plyr::count(telomerase_scores, c("broad_histology", "molecular_subtype"))
                                 broad_histology                molecular_subtype freq
1  Diffuse astrocytic and oligodendroglial tumor                      DMG, H3 K28   10
2  Diffuse astrocytic and oligodendroglial tumor           DMG, H3 K28, TP53 loss   21
3  Diffuse astrocytic and oligodendroglial tumor                 HGG, H3 wildtype   32
4  Diffuse astrocytic and oligodendroglial tumor HGG, H3 wildtype, TP53 activated    5
5  Diffuse astrocytic and oligodendroglial tumor      HGG, H3 wildtype, TP53 loss   18
6                                Embryonal tumor               CNS Embryonal, NOS   10
7                                Embryonal tumor                       MB, Group3   14
8                                Embryonal tumor                       MB, Group4   50
9                                Embryonal tumor                          MB, SHH   30
10                               Embryonal tumor                          MB, WNT   10
11                               Ependymal tumor                     EPN, ST RELA   29
12                    Low-grade astrocytic tumor                  GNG, BRAF V600E   12
13                    Low-grade astrocytic tumor               GNG, KIAA1549-BRAF    5
14                    Low-grade astrocytic tumor                    GNG, wildtype   14
15                    Low-grade astrocytic tumor                  LGG, BRAF V600E   26
16                    Low-grade astrocytic tumor        LGG, BRAF V600E, CDKN2A/B    5
17                    Low-grade astrocytic tumor                        LGG, FGFR    8
18                    Low-grade astrocytic tumor               LGG, KIAA1549-BRAF  114
19                    Low-grade astrocytic tumor                LGG, NF1-germline    6
20                    Low-grade astrocytic tumor                  LGG, other MAPK   12
21                    Low-grade astrocytic tumor                         LGG, RTK   12
22                    Low-grade astrocytic tumor                    LGG, wildtype   42
23          Mesenchymal non-meningothelial tumor                              EWS   11
24                       Tumors of sellar region                     CRANIO, ADAM   24
```

This still won't work well with your suggested palette because Low-grade astrocytic tumor is still at 11 molecular subtypes.

2. In your example for TP53, you have very few groups so it does not look messy when you add adjusted p-values on top of the boxplots. In this case, LGG has 11 subtypes (after the above filtering) so it is very messy. One solution is to keep only `p-significance` or `p-value` added by **comparing across all groups** (the way I have it currently in the code). It will just show a p-value or significance code on top of each box when compared to the rest of the boxes. Much cleaner in my opinion. 
![EXTENDScores_Low-grade_astrocytic_tumor](https://user-images.githubusercontent.com/7000615/130357369-b2e8970d-5d1e-45ed-a787-d47fc89d1564.png)

3. After all the filtering, `Mesenchymal non-meningothelial tumor` and `Tumors of sellar region` have only 1 molecular subtype - do you still want me to plot those? Of course, there won't be any p-values associated with those plots.

Reply:
Just an example, for point 2) here is what I meant:

For a much cleaner look, [calculate adjusted p-value for each subtype against "all" and add p-value significance symbols for adjusted p-values](https://github.com/komalsrathi/OpenPBTA-analysis/blob/telact-by-subtype/analyses/telomerase-activity-prediction/util/boxplot_by_molecular_subtype.R#L31-L41)  + [Use those symbols on top of each box](https://github.com/komalsrathi/OpenPBTA-analysis/blob/telact-by-subtype/analyses/telomerase-activity-prediction/util/boxplot_by_molecular_subtype.R#L59). 

![EXTENDScores_Low-grade_astrocytic_tumor](https://user-images.githubusercontent.com/7000615/130359503-40b69f7f-79f8-484e-a1ab-efe96b043cf3.png)

Reply:
oh yes - there is also a way to not add a bar if not significant - you can do that as well because I think `ns` everywhere is also a little busy

Reply:
> This still won't work well with your suggested palette because Low-grade astrocytic tumor is still at 11 molecular subtypes. (see plot below)

This is why you don't see a box for RTK.